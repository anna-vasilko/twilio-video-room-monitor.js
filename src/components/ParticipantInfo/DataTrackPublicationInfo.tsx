import React from 'react';
import { LocalDataTrack, LocalDataTrackPublication, RemoteDataTrack, RemoteDataTrackPublication } from 'twilio-video';
import useIsTrackEnabled from '../../hooks/useIsTrackEnabled/useIsTrackEnabled';
import useIsTrackSwitchedOff from '../../hooks/useIsTrackSwitchedOff/useIsTrackSwitchedOff';
import useTrack from '../../hooks/useTrack/useTrack';
import DataTrack from '../typography/DataTrack/DataTrack';
import Datum from '../typography/Datum/Datum';
import StatsContainer from '../typography/StatsContainer/StatsContainer';

export const DataTrackInfo: React.FC<{
  track: LocalDataTrack | RemoteDataTrack;
}> = ({ track }) => {
  const isEnabled = useIsTrackEnabled(track);
  const isSwitchedOff = useIsTrackSwitchedOff(track);

  if (track.isEnabled === undefined || track.isSwitchedOff === undefined) {
    return <DataTrack track={track} />;
  }

  return (
    <>
      <Datum label="isEnabled" value={String(isEnabled)} />
      <Datum label="isSwitchedOff" value={String(isSwitchedOff)} />
      <Datum label="Priority" value={track.priority} />
      <DataTrack track={track} />
    </>
  );
};

export const DataTrackPublicationInfo: React.FC<{
  publication: LocalDataTrackPublication | RemoteDataTrackPublication;
}> = ({ publication }) => {
  const track = useTrack(publication) as LocalDataTrack | RemoteDataTrack | undefined;

  return (
    <StatsContainer>
      <Datum label="Kind" value={publication.kind} />
      <Datum label="isSubscribed" value={String(!!track)} />
      {track && <DataTrackInfo track={track} />}
    </StatsContainer>
  );
};
