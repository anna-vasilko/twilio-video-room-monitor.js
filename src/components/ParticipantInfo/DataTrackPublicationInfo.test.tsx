import React from 'react';
import { shallow } from 'enzyme';
import { DataTrackPublicationInfo, DataTrackInfo } from './DataTrackPublicationInfo';
// import useIsTrackEnabled from '../../hooks/useIsTrackEnabled/useIsTrackEnabled';
// import useIsTrackSwitchedOff from '../../hooks/useIsTrackSwitchedOff/useIsTrackSwitchedOff';
import useTrack from '../../hooks/useTrack/useTrack';

jest.mock('../../hooks/useIsTrackEnabled/useIsTrackEnabled', () => () => true);
jest.mock('../../hooks/useIsTrackSwitchedOff/useIsTrackSwitchedOff', () => () => true);
jest.mock('../../hooks/useTrack/useTrack');

const mockUseTrack = useTrack as jest.Mock<any>;

mockUseTrack.mockImplementation(() => 'testTrack');

describe('the DataTrackInfo component', () => {
  it('should render correctly if an data track is present', () => {
    const wrapper = shallow(<DataTrackInfo track={{} as any} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly if there is no data track data', () => {
    const wrapper = shallow(<DataTrackInfo track={{} as any} />);
    expect(wrapper.find({ isEnabled: true }).exists()).toBeFalsy();
  });
});

describe('the DataTrackPublicationInfo component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <DataTrackPublicationInfo publication={{ trackName: 'testName', trackSid: 'testSid' } as any} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render the DataTrackInfo component if an data track is not present', () => {
    mockUseTrack.mockImplementationOnce(() => null);
    const wrapper = shallow(
      <DataTrackPublicationInfo publication={{ trackName: 'testName', trackSid: 'testSid' } as any} />
    );
    expect(wrapper.find(DataTrackInfo).exists()).toBeFalsy();
  });
});
