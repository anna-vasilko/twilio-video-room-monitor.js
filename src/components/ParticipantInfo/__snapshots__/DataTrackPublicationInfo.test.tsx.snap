// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the DataTrackInfo component should render correctly if an data track is present 1`] = `
<Fragment>
  <Memo(Datum)
    label="isEnabled"
    value="true"
  />
  <Memo(Datum)
    label="isSwitchedOff"
    value="true"
  />
  <Memo(Datum)
    label="Priority"
  />
  <Memo(Tracks)
    track={Object {}}
  />
</Fragment>
`;

exports[`the DataTrackPublicationInfo component should render correctly 1`] = `
<Memo(styled.div)>
  <Memo(Datum)
    label="Kind"
  />
  <Memo(Datum)
    label="isSubscribed"
    value="true"
  />
  <DataTrackInfo
    track="testTrack"
  />
</Memo(styled.div)>
`;
